scalameta/io/shared/src/main/scala/scala/meta/io/AbsolutePath.scala

```diff
@@ -1,29 +1,24 @@
 package scala.meta.io
 
-import scala.meta.internal.io.PlatformIO._
+import java.io._
+import org.scalameta.data._
+import scala.meta.internal.io.PlatformIO
 
-// sealed abstract to ensure that AbsolutePath.apply constructor is used.
-sealed abstract class AbsolutePath(val absolute: String) extends Serializable {
-  def relative: String = absolute.stripPrefix(workingDirectory.absolute + fileSeparator)
-  override def hashCode(): Int = absolute.hashCode
-  override def equals(obj: scala.Any): Boolean = obj match {
-    case p: AbsolutePath => absolute == p.absolute
-    case _ => false
-  }
-  override def toString: String = s"AbsolutePath($absolute)"
-  @deprecated("Use .absolute instead", "1.8.0") // kept for source compabilitity with old Input.File/Address.File
-  def getAbsolutePath: String = absolute
-  def /(other: String): AbsolutePath =
-    new AbsolutePath(absolute + fileSeparator + other.stripPrefix(fileSeparator)) {}
+@data class AbsolutePath(underlying: String) {
```

Comment:
I'm fine with a non-symbolic method name like `resolve` instead of `/`, that would be consistent with `nio.Path` and if someone really wants `/` then it's possible to with extensions (maybe via contrib).